languages/ruby/exercises/concept/instance-variables/.docs/after.md

```diff
@@ -0,0 +1,63 @@
+## Key Points:
+
+- When a class' `.new` method is called to create an object instance, the `.initialize` method is passed all arguments to initialize the instance's state.
+- instance variable names are prefixed with `@`.
+- instance variables default to `nil` until they are explicitly set.
+- instance variables are private by default, and they should be manipulated with getters and setters
+
+```ruby
+class Backpack
+  initialize(owner)
+    @owner = owner
+  end
+
+  def owner
+    @owner
+  end
+
+  def owner=(new_owner)
+    @owner = new_owner
+  end
+end
+```
+
+- Methods named with a trailing `=` are recognized as setters by ruby, and allow the optional use of the assignment syntax, e.g. `Backpack.new("Sven").owner = "Ayah"`
+- Getters and setters can be shortened using the `attr_reader`, `attr_writer`, and `attr_accessor` methods:
+  - `attr_reader`: Create getters for the symbols listed
+  - `attr_writer`: Create setters for the symbols listed
+  - `attr_accessor`: Create getters and setters for the symbols listed
+
+```ruby
+class Backpack
+  attr_accessor :owner
+
+  initialize(owner)
+    @owner = owner
+  end
+end
+```
+
+- Why use getters and setters rather than the instance variable directly?
+  - If there was a typo in the previous example (e.g. `@ownar`), it would fail silently, potentially introducing a bug into the system.
```

Comment:
Why would it fail silently at this point?  What might we get instead later on down the line?

Reply:
The expectation of the coder would be that the state of `@owner` would be updated.  Rather than the notified of the error (failing explicitly, like calling an undefined method), it fails silently, updating `@ownar` instead.

Reply:
Potentially raising confusing errors later, such as "NoMethodError: Nil can't do `whatever`".  And in an area that may not even be close to where the real problem is (The typo).